internal/inventory/app/routes.go

```diff
@@ -59,90 +60,95 @@ func (app *InventoryApp) addRoutes() error {
 }
 
-func (app *InventoryApp) addRoute(path, method string, f http.HandlerFunc) error {
-	if err := app.service.AddRoute(path, f, method); err != nil {
+func (app *InventoryApp) addRoute(path, method string, f echo.HandlerFunc) error {
+	if err := app.service.AddCustomRoute(path, false, f, method); err != nil {
 		return fmt.Errorf("failed to add route, path=%s, method=%s: %w", path, method, err)
 	}
 	return nil
 }
 
 // Routes
-func (app *InventoryApp) index(w http.ResponseWriter, req *http.Request) {
-	http.ServeFile(w, req, "static/html/index.html")
+func (app *InventoryApp) index(ctx echo.Context) error {
+	http.ServeFile(ctx.Response().Writer, ctx.Request(), "static/html/index.html")
+	return nil
 }
 
-func (app *InventoryApp) getReaders(w http.ResponseWriter, _ *http.Request) {
+func (app *InventoryApp) getReaders(ctx echo.Context) error {
+	w := ctx.Response().Writer
 	w.Header().Set("Content-Type", "application/json")
 	if err := app.defaultGrp.WriteReaders(w); err != nil {
 		msg := fmt.Sprintf("Failed to write readers list: %v", err)
 		app.lc.Error(msg)
-		w.WriteHeader(http.StatusInternalServerError)
-		http.Error(w, msg, http.StatusInternalServerError)
+		ctx.String(http.StatusInternalServerError, msg)
 	}
+	return nil
 }
 
-func (app *InventoryApp) getSnapshot(w http.ResponseWriter, _ *http.Request) {
+func (app *InventoryApp) getSnapshot(ctx echo.Context) error {
+	w := ctx.Response().Writer
 	w.Header().Set("Content-Type", "application/json")
 	if err := app.requestInventorySnapshot(w); err != nil {
 		msg := fmt.Sprintf("Failed to write inventory snapshot: %v", err)
 		app.lc.Error(msg)
-		w.WriteHeader(http.StatusInternalServerError)
-		http.Error(w, msg, http.StatusInternalServerError)
+		ctx.String(http.StatusInternalServerError, msg)
 	}
+	return nil
 }
 
-func (app *InventoryApp) startReading(w http.ResponseWriter, _ *http.Request) {
+func (app *InventoryApp) startReading(ctx echo.Context) error {
 	if err := app.defaultGrp.StartAll(app.devService); err != nil {
 		msg := fmt.Sprintf("Failed to StartAll: %v", err)
 		app.lc.Error(msg)
-		w.WriteHeader(http.StatusInternalServerError)
-		http.Error(w, msg, http.StatusInternalServerError)
+		ctx.String(http.StatusInternalServerError, msg)
 	}
+	return nil
 }
 
-func (app *InventoryApp) stopReading(w http.ResponseWriter, _ *http.Request) {
+func (app *InventoryApp) stopReading(ctx echo.Context) error {
 	if err := app.defaultGrp.StopAll(app.devService); err != nil {
 		msg := fmt.Sprintf("Failed to StopAll: %v", err)
 		app.lc.Error(msg)
-		w.WriteHeader(http.StatusInternalServerError)
-		http.Error(w, msg, http.StatusInternalServerError)
-		return
+		ctx.String(http.StatusInternalServerError, msg)
 	}
+	return nil
 }
 
-func (app *InventoryApp) getBehavior(w http.ResponseWriter, req *http.Request) {
-	rv := mux.Vars(req)
+func (app *InventoryApp) getBehavior(ctx echo.Context) error {
+	w := ctx.Response().Writer
+	rv := mux.Vars(ctx.Request())
 	bName := rv["name"]
```

Comment:
This need to be done using Echo package. Doing so will remove the use of Mux package in go.mod.

see https://github.com/edgexfoundry/edgex-go/blob/main/internal/core/data/controller/http/event.go#L75-L79

the route constant use of "/{name}" changes to "/:name" for Echo
Here: https://github.com/edgexfoundry/app-rfid-llrp-inventory/blob/main/internal/inventory/app/routes.go#L26